crates/core/src/execution.rs

```diff
@@ -37,6 +46,7 @@ pub fn run_bytecode(runtime: &Runtime, bytecode: &[u8]) {
 /// the target function from a previously evaluated module. It's the caller's
 /// reponsibility to ensure that the module containing the target function has
 /// been previously evaluated.
+#[allow(dead_code)] // Used by `main.rs` but not by `lib.rs`.
```

Comment:
Any reason why we can't get rid of this function completely? I experimented a bit on top of your changes and it seems to work and have the same performance benefits as in the dynamically linked case:

```diff
diff --git a/crates/core/src/main.rs b/crates/core/src/main.rs
index 5f8db9f..1e8eb14 100644
--- a/crates/core/src/main.rs
+++ b/crates/core/src/main.rs
@@ -66,7 +66,7 @@ pub unsafe extern "C" fn invoke(fn_name_ptr: *mut u8, fn_name_size: usize) {
 
     let js_fn_name = str::from_utf8_unchecked(slice::from_raw_parts(fn_name_ptr, fn_name_size));
     let runtime = unsafe { RUNTIME.take().unwrap() };
-    execution::invoke_function(&runtime, FUNCTION_MODULE_NAME, js_fn_name);
+    execution::run_bytecode(&runtime, &BYTECODE.take().unwrap(), Some(js_fn_name));
 }
 
 // RAII abstraction for calling Wasm ctors and dtors for exported non-main functions.
```
I see some failures, all of them related to fuel consumption being lower than expected:

```
thread 'compile_test_exported_functions' panicked at crates/cli/tests/integration_test.rs:323:5:
fuel_consumed (63156) was not within 10.00% of the target_fuel value (80023)

```

Reply:
I was a little concerned that there might be a change in behaviour or a regression in fuel since the existing implementation wasn't evaluating the bytecode. But trying it out, it works and is faster, so let's go with it!